runtimes/core/src/pubsub/nsq/sub.rs

```diff
@@ -37,8 +38,15 @@ impl NsqSubscription {
         cfg: &pb::PubSubSubscription,
         meta: &meta::pub_sub_topic::Subscription,
     ) -> Self {
-        let topic = NSQTopic::new(cfg.topic_cloud_name.clone()).unwrap();
-        let channel = NSQChannel::new(cfg.subscription_cloud_name.clone()).unwrap();
+        let topic = NSQTopic::new(&cfg.topic_cloud_name).unwrap_or_else(|| {
+            let topic_name = hash_name(&cfg.topic_cloud_name);
```

Comment:
We shouldn't handle this in the runtime, but when we generate NSQ names in the runtime config gen